apps/vast/pyvast_threatbus.py

```diff
@@ -565,6 +624,14 @@ def main():
         default=None,
         help="If sink is specified, sightings are not reported back to Threat Bus. Instead, the context of a sighting (only the contents without the Threat Bus specific sighting structure) is forwarded to the specified sink via a UNIX pipe. This option takes a command line string to use and invokes it as direct subprocess without shell / globbing support.",
     )
+    parser.add_argument(
+        "--ulimit",
+        "-U",
+        dest="ulimit",
+        default=100,
+        type=int,
+        help="Controls the maximum number of concurrent background tasks for VAST queries. Default is 100.",
```

Comment:
To me, `ulimit` means "resource limits at the process level", and usually takes an argument of the specific aspect to control. Also, the argument to `ulimit` seems to be missing (`ulimit -what?`). Is there a reason why we don't call this parameter simply what it is? Like `max_concurrent_queries` or `max_tasks`?

Reply:
My understanding of `ulimit` is that it describes the number of open file descriptors. We can make it more descriptive, though, if you think it is not clear this way :)

Reply:
Yeah, let's adapt this before shipping. The number of file descriptor is only one aspect you can control with the shell builtin `ulimit`. See the man page:

```
              Not  all  the following resources are supported on all systems.  Running ulimit -a will show which are
              supported.

              -a     Lists all of the current resource limits.
              -b     Socket buffer size in bytes (N.B. not kilobytes)
              -c     512-byte blocks on the size of core dumps.
              -d     Kilobytes on the size of the data segment.
              -f     512-byte blocks on the size of files written.
              -i     The number of pending signals.
              -k     The number of kqueues allocated.
              -l     Kilobytes on the size of locked-in memory.
              -m     Kilobytes on the size of physical memory.
              -n     open file descriptors.
              -p     The number of pseudo-terminals.
              -q     Bytes in POSIX message queues.
              -r     Maximum real time priority.  On some systems where this is not available, such as NetBSD,  this
                     has the same effect as -T for compatibility with sh.
              -s     Kilobytes on the size of the stack.
              -T     The number of simultaneous threads available to the user.
              -t     CPU seconds to be used.
              -u     The number of processes available to the user.
              -v     Kilobytes  on  the  size  of  virtual  memory.  On some systems this refers to the limit called
                     `address space'.
              -w     Kilobytes on the size of swapped out memory.
              -x     The number of locks on files.
```

The `-n` flag is for file descriptors.

Reply:
I renamed the config option.